pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -229,7 +229,9 @@ synchronized private void setup() throws Exception {
         setupLogHandler();
 
         javaInstance = new JavaInstance(contextImpl, object, instanceConfig);
-
+        Thread.currentThread().setContextClassLoader(functionClassLoader);
+        javaInstance.setup();
+        Thread.currentThread().setContextClassLoader(instanceClassLoader);
```

Comment:
We should do this I  a finally block

Reply:
Sorry I did not catch this.
If exception happened in user's setup() code,  `JavaInstanceRunnable.run()` will catch the exception and close function instance. Why we need to do this in a finally block

Reply:
We need to make to reset the class loader to the `instanceClassLoader`.

Reply:
> We need to make sure to reset the class loader to the `instanceClassLoader`.

+1

Reply:
done